src/Higgs4LeptonsFinder.cc

```diff
@@ -10,47 +10,51 @@ bool Higgs4LeptonsFinder::process(HiggsFourLeptonsEvent& event) {
-      const Electron lep2 = (*event.electrons).at(j);
+  for(size_t i=0; i<(*event.muons).size(); i++){
+    const Muon lep1 = (*event.muons).at(i);
+    for(size_t j=i+1; j<(*event.muons).size(); j++){
+      const Muon lep2 = (*event.muons).at(j);
       if (lep1.charge()==lep2.charge()) continue;
       TLorentzVector z1 = lep1.p4()+lep2.p4();
       if (Fail_ZMassMax(z1)) continue;
       if (Fail_Z2MassMin(z1)) continue;
       if ( FindInVector<int>(lep_indices, i)<0 ) {
         lep_indices.push_back(i);
-        lep_bool.push_back(true);
+        lep_bool.push_back(false);
       }
       if ( FindInVector<int>(lep_indices, j)<0 ) {
         lep_indices.push_back(j);
-        lep_bool.push_back(true);
+        lep_bool.push_back(false);
       }
     }
   }
 
-  for(size_t i=0; i<(*event.muons).size(); i++){
-    const Muon lep1 = (*event.muons).at(i);
-    for(size_t j=i+1; j<(*event.muons).size(); j++){
-      const Muon lep2 = (*event.muons).at(j);
+  for(size_t i=0; i<(*event.electrons).size(); i++){
+    const Electron lep1 = (*event.electrons).at(i);
+    if(Fail_CrossCleaning(lep1,(*event.muons),lep_bool,lep_indices)) continue;
+    for(size_t j=i+1; j<(*event.electrons).size(); j++){
+      const Electron lep2 = (*event.electrons).at(j);
       if (lep1.charge()==lep2.charge()) continue;
       TLorentzVector z1 = lep1.p4()+lep2.p4();
       if (Fail_ZMassMax(z1)) continue;
       if (Fail_Z2MassMin(z1)) continue;
-      if ( std::find(lep_indices.begin(), lep_indices.end(), i) == lep_indices.end() || (std::find(lep_indices.begin(), lep_indices.end(), i) != lep_indices.end() && lep_bool.at(std::find(lep_indices.begin(), lep_indices.end(), i) - lep_indices.begin())) ) {
+      if(Fail_CrossCleaning(lep2,(*event.muons),lep_bool,lep_indices)) continue;
+      if ( std::find(lep_indices.begin(), lep_indices.end(), i) == lep_indices.end() || (std::find(lep_indices.begin(), lep_indices.end(), i) != lep_indices.end() && !lep_bool.at(std::find(lep_indices.begin(), lep_indices.end(), i) - lep_indices.begin())) ) {
         lep_indices.push_back(i);
-        lep_bool.push_back(false);
+        lep_bool.push_back(true);
       }
-      if ( std::find(lep_indices.begin(), lep_indices.end(), j) == lep_indices.end() || (std::find(lep_indices.begin(), lep_indices.end(), j) != lep_indices.end() && lep_bool.at(std::find(lep_indices.begin(), lep_indices.end(), j) - lep_indices.begin())) ) {
+      if ( std::find(lep_indices.begin(), lep_indices.end(), j) == lep_indices.end() || (std::find(lep_indices.begin(), lep_indices.end(), j) != lep_indices.end() && !lep_bool.at(std::find(lep_indices.begin(), lep_indices.end(), j) - lep_indices.begin())) ) {
         lep_indices.push_back(j);
-        lep_bool.push_back(false);
+        lep_bool.push_back(true);
       }
     }
   }
+  
 
   if (lep_indices.size()<4) return false;
 
   vector<vector<int>> used_indices={};
+  vector<FlavorParticle> choices_lep4 = {};
```

Comment:
this is not needed

Reply:
right
used_indices is also useless isn't it ? Because we store the indices at line 104 but we don't use them. Them it would be the same with the variable "indices" defined at line 82. Now that we select according to chi square we don't need to store the indice set right ?

Reply:
Correct. that can also go

Reply:
Both were removed